components/kubernetes/Apply_object/component.yaml

```diff
@@ -0,0 +1,34 @@
+name: Apply Kubernetes object
+inputs:
+- {name: Object, type: JsonObject}
+outputs:
+- {name: Name, type: String}
+- {name: Kind, type: String}
+- {name: Object, type: JsonObject}
+metadata:
+  annotations:
+    author: Alexey Volkov <alexey.volkov@ark-kun.com>
+implementation:
+  container:
+    image: bitnami/kubectl:1.17.17
+    command:
+    - bash
+    - -exc
+    - |
+      object_path=$0
+      output_name_path=$1
+      output_kind_path=$2
+      output_object_path=$3
+      mkdir -p "$(dirname "$output_name_path")"
+      mkdir -p "$(dirname "$output_kind_path")"
+      mkdir -p "$(dirname "$output_object_path")"
+
+      kubectl apply -f "$object_path" --output=json > "$output_object_path"
+      
+      < "$output_object_path" jq '.metadata.name' --raw-output > "$output_name_path"
+      < "$output_object_path" jq '.kind' --raw-output > "$output_kind_path"
+
+    - {inputPath: Object}
+    - {outputPath: Name}
+    - {outputPath: Kind}
+    - {outputPath: Object}
```

Comment:
What is the difference between `create` and `apply` objects?

Reply:
The difference is pretty small. Create always tries to create a new object and fails when it already exists.
Apply patches or replaces an object if it exists. Create supports auto-generating object names (`generateName`), but apply does not.
I'm adding both for the sake of completeness. There is a long-term plan to replace or retrofit `kfp.dsl.ResourceOp` with these components.

Reply:
Thank you Alexey! Do we plan to support `generateName` for Create, and fails if already exists in current PR? Or as you mentioned, it is for future plan?

Reply:
`generateName` is supported as part of the Kubernetes object passed to the `Object` input.
If you pass structure that has:
```yaml
...
metadata:
    generateName: "some-prefix-"
...
```
to the "Create Kubernetes object" component, which calls `kubectl create` then the object name would be generated by Kubernetes and returned from the component.

If you pass that to the "Apply Kubernetes object" component, then `kubectl` will fail and the component will fail as it should.

Reply:
I see, thank you Alexey!